test/unit/BPool/BPool_JoinPool.t.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {BPoolBase} from './BPoolBase.sol';
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+import {BNum} from 'contracts/BNum.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+
+contract BPoolJoinPool is BPoolBase {
+  uint256 public poolAmountOut = 10e18;
```

Comment:
what if you define `poolAmountOut` as `INIT_POOL_SUPPLY / x` and then explain that you expect required token amounts related to the initial balance and `x`